internal/operands/template-validator/reconcile_test.go

```diff
@@ -83,6 +88,207 @@ var _ = Describe("Template validator operand", func() {
+								Values:   []string{"true"},
+							},
+						},
+					},
+					TopologyKey: "kubernetes.io/hostname",
+				},
+			},
+		}
+		defaultPodAntiAffinity := &core.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []core.WeightedPodAffinityTerm{
+				{
+					Weight: 1,
+					PodAffinityTerm: core.PodAffinityTerm{
+						LabelSelector: &meta.LabelSelector{
+							MatchExpressions: []meta.LabelSelectorRequirement{
+								{
+									Key:      "kubevirt.io",
+									Operator: meta.LabelSelectorOpIn,
+									Values:   []string{"virt-template-validator"},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+		}
+		mergedPodAntiAffinity := &core.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []core.WeightedPodAffinityTerm{
+				{
+					Weight: 1,
+					PodAffinityTerm: core.PodAffinityTerm{
+						LabelSelector: &meta.LabelSelector{
+							MatchExpressions: []meta.LabelSelectorRequirement{
+								{
+									Key:      "kubevirt.io",
+									Operator: meta.LabelSelectorOpIn,
+									Values:   []string{"virt-template-validator"},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+				{
+					Weight: 1,
+					PodAffinityTerm: core.PodAffinityTerm{
+						LabelSelector: &meta.LabelSelector{
+							MatchExpressions: []meta.LabelSelectorRequirement{
+								{
+									Key:      "prefAntiAffinityKey",
+									Operator: meta.LabelSelectorOpIn,
+									Values:   []string{"true"},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+			RequiredDuringSchedulingIgnoredDuringExecution: []core.PodAffinityTerm{
+				{
+					LabelSelector: &meta.LabelSelector{
+						MatchExpressions: []meta.LabelSelectorRequirement{
+							{
+								Key:      "reqAntiAffinityKey",
+								Operator: meta.LabelSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+					TopologyKey: "kubernetes.io/hostname",
+				},
+			},
+		}
+
+		var setPodAntiAffinity = func(request *common.Request) {
+			request.Instance.Spec.TemplateValidator.Placement.Affinity.PodAntiAffinity = antiAffinity
+			return
+		}
+		var setPodAffinity = func(request *common.Request) {
+			request.Instance.Spec.TemplateValidator.Placement.Affinity.PodAffinity = podAffinity
+			return
+		}
+		var setNodeAffinity = func(request *common.Request) {
+			request.Instance.Spec.TemplateValidator.Placement.Affinity.NodeAffinity = nodeAffinity
+			return
+		}
+
+		table.DescribeTable("should create correct deployment", func(requestAdjustFunctions []func(*common.Request), expectedNodeAffinity *core.NodeAffinity, expectedPodAffinity *core.PodAffinity, expectedPodAntiAffinity *core.PodAntiAffinity) {
+			for _, f := range requestAdjustFunctions {
+				f(&request)
+			}
+			res, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+			deploymentResult := res[4]
```

Comment:
Can you use the client to get the deployment instead of accessing the result of the call? The resource in result is only used for getting names and logging.

You can use:
```go
deployment := &apps.Deployment{}
key := client.ObjectKeyFromObject(newDeployment(namespace, replicas, "test-img"))
Expect(request.Client.Get(request.Context, key, deployment)).To(Succeed())
```